### PR TITLE
[scarb-doc] Scarb packages

### PR DESCRIPTION
**Pull request by drknzz**

### CODE REVIEW

**Comment by maciektr** [comment]
Pls rebase

File: extensions/scarb-doc/src/main.rs [hunk 1]
```diff
@@ -53,12 +74,135 @@ fn main() -> Result<()> {
...
+        .expect("Corelib could not be found")
+}
+
+fn get_dependencies(
+    compilation_unit_metadata: &CompilationUnitMetadata,
+) -> OrderedHashMap<SmolStr, PathBuf> {
+    compilation_unit_metadata
+        .components
+        .iter()
+        .filter(|du| du.name != "core")
```
Comment (maciektr) [inline extensions/scarb-doc/src/main.rs:131]:
```suggestion
        .filter(|du| du.name != CORELIB_CRATE_NAME)
```

File: extensions/scarb-doc/src/main.rs [hunk 2]
```diff
@@ -53,12 +74,135 @@ fn main() -> Result<()> {
...
+    metadata: &Metadata,
+    compilation_unit_metadata: &CompilationUnitMetadata,
+) -> AllCratesConfig {
+    let crates_config: OrderedHashMap<SmolStr, CrateSettings> = compilation_unit_metadata
+        .components
+        .iter()
+        .map(|component| {
+            let pkg = metadata.get_package(&component.package).unwrap_or_else(|| {
+                panic!(
+                    "Failed to find = {} package",
```
Comment (maciektr) [inline extensions/scarb-doc/src/main.rs:151]: I would suggest using scarb style of error messages here as well, so please don't start with capital letters 
```suggestion
                    "failed to find `{}` package",
```

File: extensions/scarb-doc/src/main.rs [hunk 3]
```diff
@@ -53,12 +74,135 @@ fn main() -> Result<()> {
...
         println!("{:?}: {:?} -> {:?}", name, item_id, doc);
     }
 
     Ok(())
 }
+
+fn get_project_config(metadata: &Metadata, package_metadata: &PackageMetadata) -> ProjectConfig {
+    let compilation_unit_metadata =
+        package_lib_compilation_unit(metadata, package_metadata.id.clone())
+            .expect("Failed to find compilation unit for package");
```
Comment (maciektr) [inline extensions/scarb-doc/src/main.rs:90]:
```suggestion
            .expect("failed to find compilation unit for `{}` package", package_metadata.id.clone());
```

**Review by maciektr — CHANGES_REQUESTED** [review]

**Review by maciektr — APPROVED** [review]
lgtm!

**Review by Arcticae — COMMENTED** [review]
Can it select different crates from a workspace? 
I think this case should be tested

File: extensions/scarb-doc/src/main.rs [hunk 4]
```diff
@@ -1,30 +1,51 @@
...
 
 use cairo_lang_defs::db::DefsGroup;
 use cairo_lang_defs::ids::{LookupItemId, ModuleItemId, NamedLanguageElementId};
 
+use cairo_lang_filesystem::ids::Directory;
 use cairo_lang_semantic::db::SemanticGroup;
 
 #[derive(Parser, Debug)]
 #[command(version, about, long_about = None)]
 struct Args {
```
[the remark this answers is not in the document]
Comment (drknzz) [reply extensions/scarb-doc/src/main.rs:26]: Already added 👍

File: extensions/scarb-doc/tests/test.rs [hunk 5]
```diff
@@ -36,3 +108,57 @@ fn test_main() {
...
+
+    let output = Scarb::quick_snapbox()
+        .arg("doc")
+        .arg("-p")
+        .arg("goodbye_world")
+        .current_dir(&t)
+        .assert()
+        .success();
+    let stdout = std::str::from_utf8(&output.get_output().stdout).unwrap();
+    assert!(stdout.contains("/// Goodbye world"));
```
Comment (maciektr) [inline extensions/scarb-doc/tests/test.rs:163]:
```suggestion
    assert!(stdout.contains("/// Goodbye world"));
    assert!(!stdout.contains("/// Hello world"));
```

**Review by maciektr — APPROVED** [review]

**Pull request merged**